core/src/main/java/lucee/transformer/bytecode/statement/tag/TagProperty.java

```diff
@@ -0,0 +1,195 @@
+		String hint = null;
+		String displayname = null;
+		boolean required = false;
+		boolean setter = true;
+		boolean getter = true;
+
+		// Parse tag attributes
+		Iterator<Attribute> it = tag.getAttributes().values().iterator();
+		while (it.hasNext()) {
+			Attribute attr = it.next();
+			String attrName = attr.getName().toLowerCase();
+
+			if ("name".equals(attrName)) {
+				name = getLiteralString(attr);
+			}
+			else if ("type".equals(attrName)) {
+				type = getLiteralString(attr);
+			}
+			else if ("default".equals(attrName)) {
+				// Handle default value - simplified for now
+				defaultValue = getLiteralString(attr);
+			}
+			else if ("access".equals(attrName)) {
+				access = getLiteralString(attr);
+			}
+			else if ("hint".equals(attrName)) {
+				hint = getLiteralString(attr);
+			}
+			else if ("displayname".equals(attrName)) {
+				displayname = getLiteralString(attr);
+			}
+			else if ("required".equals(attrName)) {
+				String val = getLiteralString(attr);
+				required = "yes".equalsIgnoreCase(val) || "true".equalsIgnoreCase(val);
+			}
+			else if ("setter".equals(attrName)) {
+				String val = getLiteralString(attr);
+				setter = !"no".equalsIgnoreCase(val) && !"false".equalsIgnoreCase(val);
+			}
+			else if ("getter".equals(attrName)) {
+				String val = getLiteralString(attr);
+				getter = !"no".equalsIgnoreCase(val) && !"false".equalsIgnoreCase(val);
+			}
+		}
+
+		System.err.println("[DEBUG-PROPERTY] Generating optimized call: ComponentUtil.registerProperty(pc, name=\"" + name + "\", type=\"" + type + "\")");
+
+		// Generate call: ComponentUtil.registerProperty(pc, name, type, defaultValue, access, hint, displayname, required, setter, getter);
+		adapter.loadArg(0); // PageContext pc
+
+		// name parameter
+		if (name != null) {
+			adapter.push(name);
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// type parameter
+		if (type != null) {
+			adapter.push(type);
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// defaultValue parameter
+		if (defaultValue != null) {
+			adapter.push(defaultValue.toString());
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// access parameter
+		if (access != null) {
+			adapter.push(access);
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// hint parameter
+		if (hint != null) {
+			adapter.push(hint);
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// displayname parameter
+		if (displayname != null) {
+			adapter.push(displayname);
+		}
+		else {
+			ASMConstants.NULL(adapter);
+		}
+
+		// required parameter
+		adapter.push(required);
+
+		// setter parameter
+		adapter.push(setter);
+
+		// getter parameter
+		adapter.push(getter);
+
+		// Call ComponentUtil.registerProperty
+		Method method = new Method("registerProperty", Type.VOID_TYPE, new Type[] { Types.PAGE_CONTEXT, Types.STRING, Types.STRING, Types.OBJECT,
+				Types.STRING, Types.STRING, Types.STRING, Type.BOOLEAN_TYPE, Type.BOOLEAN_TYPE, Type.BOOLEAN_TYPE });
```

Comment:
i prefer to define Methods as static Constants in the beginning of the class and then just use the constants, that way he class get created only once and not every single time. Alos it makes it easier to read.
```
private static Method REGISTER_PROPERTY = new Method("registerProperty", Type.VOID_TYPE, new Type[] { Types.PAGE_CONTEXT, Types.STRING, Types.STRING, Types.OBJECT,
				Types.STRING, Types.STRING, Types.STRING, Type.BOOLEAN_TYPE, Type.BOOLEAN_TYPE, Type.BOOLEAN_TYPE });
...
adapter.invokeStatic(Type.getType("Llucee/runtime/type/util/ComponentUtil;"), REGISTER_PROPERTY);
```